src/services/chain.service.ts

```diff
@@ -0,0 +1,226 @@
+            const client = this._client.slice(0, this._dfractOnChainApy.length);
+
+            // Chains where inflation is retrievable
+            const getChainApy = await Promise.all(
+                client.map(async (el, index) => {
+                    const inflation = Number(await el.queryClient.mint.inflation()) / CLIENT_PRECISION;
+
+                    const metrics = await computeTotalApy(el, Number((await this.getTokenSupply())[index].supply), inflation, CLIENT_PRECISION, TEN_EXPONENT_SIX);
+
+                    return { apy: apy(metrics.inflation, metrics.communityTaxRate, metrics.stakingRatio), symbol: this._assetSymbol[index] };
+                }),
+            );
+
+            // Evmos manual inflation calculation
+            // We use their official endpoint to retrieve the inflation rate
+            const evmosIndex = CHAIN_ENV_CONFIG.findIndex((el) => el === `${AssetSymbol.EVMOS}_NETWORK_ENDPOINT`);
+            const evmosInflation =
+                Number(await lastValueFrom(this._httpService.get(`https://rest.bd.evmos.dev:1317/evmos/inflation/v1/inflation_rate`).pipe(map((response) => response.data.inflation_rate)))) /
+                PERCENTAGE;
+
+            const metrics = await computeTotalApy(this._client[evmosIndex], Number((await this.getTokenSupply())[evmosIndex].supply), evmosInflation, CLIENT_PRECISION, CLIENT_PRECISION);
+
+            const getEvmosApy = { apy: apy(metrics.inflation, metrics.communityTaxRate, metrics.stakingRatio), symbol: this._assetSymbol[evmosIndex] };
+
+            // Chains for which inflation is not retrievable via the mint module. Hence, we rely on other rpc endpoints to retrieve it.
+            const getOsmosisApy = Number(await lastValueFrom(this._httpService.get(`https://api-osmosis.imperator.co/apr/v2/staking`).pipe(map((response) => response.data)))) / PERCENTAGE;
+            const getJunoApy = Number(await lastValueFrom(this._httpService.get(`https://supply-api.junonetwork.io/apr`).pipe(map((response) => response.data)))) / PERCENTAGE;
+            const getStargazeApy = Number(await lastValueFrom(this._httpService.get(`https://supply-api.publicawesome.dev/apr`).pipe(map((response) => response.data)))) / PERCENTAGE;
+
+            return [
+                ...getChainApy,
+                getEvmosApy,
+                { apy: getOsmosisApy, symbol: AssetSymbol.OSMOSIS },
+                { apy: getJunoApy, symbol: AssetSymbol.JUNO },
+                { apy: getStargazeApy, symbol: AssetSymbol.STARGAZE },
+            ];
+        } catch (error) {
+            this._logger.error(`Could not fetch Token Apy for External Chains...`, error);
+            return null;
+        }
+    };
+
+    getAssetInfo = async (): Promise<AssetInfo[]> => {
+        try {
+            return await Promise.all([await this.getPrice(), await this.getMcap(), await this.getTokenSupply(), await this.getApy()]).then(([unit_price_usd, total_value_usd, supply, apy]) =>
```

Comment:
This should be improved and split on multiple lines, however quite hard to read through